kubernetes/structures_pod.go

```diff
@@ -279,9 +279,24 @@ func flattenEmptyDirVolumeSource(in *v1.EmptyDirVolumeSource) []interface{} {
 
 func flattenSecretVolumeSource(in *v1.SecretVolumeSource) []interface{} {
 	att := make(map[string]interface{})
+	if in.DefaultMode != nil {
+		att["default_mode"] = *in.DefaultMode
+	}
 	if in.SecretName != "" {
 		att["secret_name"] = in.SecretName
 	}
+	if len(in.Items) > 0 {
+		items := make([]interface{}, len(in.Items))
+		for i, v := range in.Items {
+			m := map[string]interface{}{}
+			m["key"] = v.Key
+			m["mode"] = int(*v.Mode)
+			m["path"] = v.Path
+			items[i] = m
+		}
+		att["items"] = items
+	}
+	att["optional"] = *in.Optional
```

Comment:
We can't assume this codepath will only ever process resources created by Terraform. Keep in mind that it's also used for imports - so even if we always pass `Optional` to the API when creating the resource, it doesn't mean it will always come back.

TL;DR we should add a nil check, IMO.